lib/brainstem/concerns/controller_dsl.rb

```diff
@@ -156,27 +207,176 @@ def model_params(root = Proc.new { |klass| klass.brainstem_model_name }, &block)
         # @option options [String,Symbol] :item_type The data type of the items contained in a field.
         #   Ideally used when the data type of the field is an `array`, `object` or `hash`.
         #
-        def valid(field_name, type = nil, options = {}, &block)
+        def valid(name, type = nil, options = {}, &block)
           valid_params = configuration[brainstem_params_context][:valid_params]
-          field_config = format_field_configuration(type, options, &block)
+          param_config = format_field_configuration(valid_params, type, options, &block)
 
-          # Inherit `nodoc` attribute from parent
-          parent_key = (options[:ancestors] || []).reverse.first
-          field_config[:nodoc] = true if parent_key && valid_params[parent_key] && valid_params[parent_key][:nodoc]
+          formatted_name = convert_to_proc(name)
+          valid_params[formatted_name] = param_config
 
-          # Rollup `required` attribute to ancestors if true
-          if field_config[:required]
-            (options[:ancestors] || []).reverse.each do |ancestor_key|
-              valid_params[ancestor_key][:required] = true if valid_params.has_key?(ancestor_key)
-            end
+          with_options(format_ancestry_options(formatted_name, param_config), &block) if block_given?
+        end
+
+        #
+        # Allows defining a custom response structure for an action.
+        #
+        # @param [Symbol] type the data type of the response.
+        # @param [Hash] options
+        # @option options [String] :info the documentation for the param
```

Comment:
I'm seeing two different styles for formatting these:

> @option [String] options :info ...

and

> @option options [String] :info ...

Do we want to keep them consistent (now or later or not)? Perhaps we should starting using some basic level of linting.